### PR TITLE
bareos: add service module for bareos

### PR DESCRIPTION
**Pull request by derchrisuk**
###### Motivation for this change

Added a new service module for bareos.
At this time only the client side has been added.
Bareos server side will be added later.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/backup/bareos.nix [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+          so that error messages can be easily identified if you have multiple
+          Clients. This directive is required.
+        '';
+      };
+
+      listen = mkOption {
+        type = with types; listOf (submodule {
+          options = {
+            protocol = mkOption {
+              type = types.nullOr types.str;
```
Comment (peterhoeg) [inline nixos/modules/services/backup/bareos.nix]: You should probably use ```types.enum``` for this.

File: nixos/modules/services/backup/bareos.nix [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+              description = ''
+                IP address client should listen on.
+              '';
+            };
+          };
+        });
+      };
+
+      compatible = mkOption {
+         default = "yes";
```
Comment (peterhoeg) [inline nixos/modules/services/backup/bareos.nix]: This should be a ```types.bool``` with a function to map true/false to yes/no.

Comment (peterhoeg) [inline nixos/modules/services/backup/bareos.nix]: This should be a ```types.bool``` with a function to map true/false to yes/no.

File: nixos/modules/services/backup/bareos.nix (hunk 1 above)
Comment (derchrisuk) [inline nixos/modules/services/backup/bareos.nix]: Good idea, will add it

File: nixos/modules/services/backup/bareos.nix (hunk 2 above)
Comment (derchrisuk) [inline nixos/modules/services/backup/bareos.nix]: Already using a function for bool mapping, can extend it to other true/false, yes/no options

**Comment by Mic92** [comment]
The module should be listed in `nixos/modules/module-list.nix`.
Also I got an evaluation error, when trying to enable it ( `listen` is defined after usage).

**Comment by derchrisuk** [comment]
Added it to the module list.
As for the evaluation error, for now, I added a default value for both listen type and address.

**Comment by bobvanderlinden** [comment]
This PR is quite old with no activity. Is there motivation to bring this PR up-to-date?

**Comment by aanderse** [comment]
@derchrisuk are you able to address to the requested changes?

**Comment by JohnAZoidberg** [comment]
Are you still interested in pursing this?
Now bareos doesn't even build anymore and the new version has changed how they build the package so a rewrite of the derivation would be necessary.

**Comment by derchrisuk** [comment]
Sorry, I don't have the time or need to look into this at the moment,

**Comment by aanderse** [comment]
If anyone is interested in picking this up feel free to open a new PR.

**Pull request closed without merging**